### PR TITLE
[Quant][fx][bc-breaking] Replace is_reference with convert_to_reference

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80091

Summary: This PR removes the is_reference flag from the  existing
convert_fx API and replaces it with a new convert_to_reference
function. This separates (1) converting the prepared model to a
reference model from (2) lowering the reference model to a quantized
model, enabling users to call their custom lowering function for
custom backends. For the native fbgemm backend, for example, the
following are equivalent:

Before (still works after this PR):
```
from torch.ao.quantization.quantize_fx import prepare_fx, convert_fx

prepared = prepare_fx(model, ...)
quantized = convert_fx(prepared, ...)
```

After:
```
from torch.ao.quantization.fx import lower_to_fbgemm
from torch.ao.quantization.quantize_fx import (
    prepare_fx,
    convert_to_reference
)

prepared = prepare_fx(model, ...)
reference = convert_to_reference(prepared, ...)
quantized = lower_to_fbgemm(reference, ...)
```

Note that currently `lower_to_fbgemm` takes in two other arguments
that are difficult for users to provide. A future commit will remove
these arguments to make the helper function more user friendly.

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

Differential Revision: [D37359946](https://our.internmc.facebook.com/intern/diff/D37359946)

### CODE REVIEW

File: torch/ao/quantization/fx/convert.py [hunk 1]
```diff
@@ -316,7 +316,11 @@ def convert_standalone_module(
         produce a reference model or a fbgemm/qnnpack model
       - backend_config_dict: backend configuration of the target backend of quantization
     """
-    convert = torch.ao.quantization.quantize_fx.convert_fx  # type: ignore[attr-defined]
+    # TODO: remove is_reference flag
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/convert.py:319]: not sure if we can do this, probably need discussion about how do we handle is_reference here in standalone modules, for example, one option is to always convert to reference quantized model and do lowering in a separate step

**Review by jerryzh168 — APPROVED** [review]
looks good! please do ghimport and fix internal callsites as well

File: torch/ao/quantization/fx/convert.py (hunk 1 above)
Comment (andrewor14) [reply torch/ao/quantization/fx/convert.py:319]: Yes, though I think we would need to refactor the lowering part out of `torch.ao.quantization.fx.convert` before we can do that

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37359946).

**Pull request closed without merging**